packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -213,10 +214,23 @@ export class BaseSnapExecutor {
 
   private async onCommandRequest(message: JsonRpcRequest) {
     if (!isJsonRpcRequest(message)) {
-      throw rpcErrors.invalidRequest({
-        message: 'Command stream received a non-JSON-RPC request.',
-        data: message,
-      });
+      if (
+        hasProperty(message, 'id') &&
+        is((message as Pick<JsonRpcRequest, 'id'>).id, JsonRpcIdStruct)
+      ) {
+        // Instead of throwing, we directly respond with an error.
+        // We can only do this if the message ID is still valid.
```

Comment:
Should we throw if the message ID is invalid?

Reply:
Throwing doesn't really do anything because the logging is turned off in production IIRC. So I'm not sure, I think I would prefer not throwing inside the executor if we can't pipe it back to MM

Reply:
Should we log a message then? So we can at least debug the problem when it occurs?